### PR TITLE
quiche: do not use Http2String wrapper for std::string

### PR DESCRIPTION
**Pull request by bencebeky**
Signed-off-by: Bence Béky <bnc@google.com>

For an explanation of how to fill out the fields, please see the relevant section
in [PULL_REQUESTS.md](https://github.com/envoyproxy/envoy/blob/master/PULL_REQUESTS.md)

Description: This is part of the effort to deprecate SpdyString and HttpString in QUICHE. QuicString has already been removed. These are/were all transparent wrappers to std::string, existing for historical reasons.

Risk Level: low

Testing: this is a change to tests

Docs Changes: n/a

Release Notes: n/a

[Optional Fixes #Issue]:
[Optional Deprecated:]

### CODE REVIEW

**Comment by mattklein123** [comment]
LGTM. Please check clang-tidy CI. Thank you.

/wait

File: test/extensions/quic_listeners/quiche/dummy_test.cc [hunk 1]
```diff
@@ -10,7 +10,7 @@ namespace Quiche {
 namespace {
 
 TEST(DummyTest, Dummy) {
-  http2::Http2String foo = "bar";
+  std::string foo = "bar";
   EXPECT_EQ("bar cowbell", moreCowbell(foo));
```
[the remark this answers is not in the document]
Comment (bencebeky) [reply test/extensions/quic_listeners/quiche/dummy_test.cc]: How about removing the entire file instead?  :)

**Comment by bencebeky** [comment]
Since CircleCI/coverage failed with something that did not seem obvious to me, and this PR started to grow into two separate directions anyway, I split off half of it into https://github.com/envoyproxy/envoy/pull/7871.  This way if the CircleCI failure is real, identifying which part causes it will help me track it down, and if it's not, well then I would have had to trigger a new run anyway.

**Pull request merged**